fractions/flyway/src/main/java/org/wildfly/swarm/flyway/runtime/FlywayMigrationArchivePreparer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.flyway.runtime;
+
+import org.wildfly.swarm.flyway.deployment.FlywayMigrationServletContextListener;
+import java.util.List;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import org.jboss.shrinkwrap.api.Archive;
+import org.wildfly.swarm.config.datasources.DataSource;
+import org.wildfly.swarm.datasources.DatasourcesFraction;
+import org.wildfly.swarm.flyway.FlywayFraction;
+import org.wildfly.swarm.spi.api.ArchivePreparer;
+import org.wildfly.swarm.undertow.descriptors.WebXmlAsset;
+import org.wildfly.swarm.undertow.descriptors.WebXmlContainer;
+
+/**
+ *
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public class FlywayMigrationArchivePreparer implements ArchivePreparer {
+
+    @Inject
+    private Instance<DatasourcesFraction> dsFractionInstance;
```

Comment:
Also, for a future iteration, @Inject @ConfigurationValue for some FlywayProperties.USERNAME and .PASSWORD.  

This would allow it to be used without having a main() configuring the FlywayFraction.  Akin to how we deal with other datasources and such.

    String thingy = this.injectedThingy.orElse( this.fraction.thingy() )

Or whatnot.  So that `project-stages.yml` or properties would suffice to configure in lieu of a `main()`.